jdaviz/configs/rampviz/plugins/ramp_extraction/ramp_extraction.py

```diff
@@ -0,0 +1,387 @@
+import numpy as np
+import astropy.units as u
+from astropy.nddata import NDDataArray
+
+from functools import cached_property
+from traitlets import Bool, Float, List, Unicode, observe
+
+from jdaviz.core.events import SnackbarMessage, SliceValueUpdatedMessage
+from jdaviz.core.marks import PluginLine
+from jdaviz.core.registries import tray_registry
+from jdaviz.core.template_mixin import (PluginTemplateMixin,
+                                        DatasetSelectMixin,
+                                        SelectPluginComponent,
+                                        ApertureSubsetSelectMixin,
+                                        ApertureSubsetSelect,
+                                        AddResultsMixin,
+                                        skip_if_not_tray_instance,
+                                        skip_if_no_updates_since_last_active,
+                                        with_spinner, with_temp_disable)
+from jdaviz.core.user_api import PluginUserApi
+from jdaviz.configs.cubeviz.plugins.viewers import WithSliceIndicator
+
+
+__all__ = ['RampExtraction']
+
+
+@tray_registry(
+    'ramp-extraction', label="Ramp Extraction", viewer_requirements='profile'
+)
+class RampExtraction(PluginTemplateMixin, ApertureSubsetSelectMixin,
+                     DatasetSelectMixin, AddResultsMixin):
```

Comment:
can this not inherit from spectral extraction?  Is there any shared logic that can be moved up?

Reply:
I'd say it shouldn't inherit from anything spectral, but we could make a `CubeExtraction` parent to both.

Reply:
I am too, let's do it. Do you want to see that in [this ticket](https://jira.stsci.edu/browse/JDAT-4618)/PR, or do you mind if I put that together with [this one](https://jira.stsci.edu/browse/JDAT-4619)?

Reply:
separate is probably fine to keep the diff somewhat manageable